src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -215,6 +218,19 @@ public FrontendEnvironment FrontendEnvironment
 
         public void AddDirective(Command command) => SubmissionParser.AddDirective(command);
 
+        public virtual void RegisterCommandHandler<TCommand>(string commandType, Func<TCommand, KernelInvocationContext, Task> handler)
+            where TCommand : KernelCommand
+        {
+            RegisterCommandType<TCommand>(commandType);
+            _dynamicHandlers[typeof(TCommand)] = (command, context) => handler((TCommand)command, context);
+        }
+
+        public void RegisterCommandType<TCommand>(string commandType)
```

Comment:
Should we derive the command name from the type for consistency?

Reply:
Good point. Done.